core/renderers/block_rendering_rewrite/block_render_draw_highlight.js

```diff
@@ -50,176 +50,176 @@ goog.require('Blockly.blockRendering.Measurable');
+    pathString =
+        Blockly.utils.svgPaths.moveTo(row.width, row.yPos) +
+        this.puzzleTabPaths_.pathDown(this.RTL_);
   }
+
+  this.steps_.push(pathString);
 };
 
 Blockly.blockRendering.Highlighter.prototype.drawStatementInput = function(row) {
-  var x = row.statementEdge;
-  if (this.info_.RTL) {
-    this.highlightSteps_.push('M',
-        (x + Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE) +
-        ',' + (row.yPos + Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE));
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.INNER_TOP_LEFT_CORNER_HIGHLIGHT_RTL);
-    this.highlightSteps_.push('v',
-        row.height - 2 * Blockly.blockRendering.constants.CORNER_RADIUS);
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.INNER_BOTTOM_LEFT_CORNER_HIGHLIGHT_RTL);
+  if (this.RTL_) {
+    var innerHeight = row.height - (2 * this.insideCornerPaths_.height);
+    var steps =
+        Blockly.utils.svgPaths.moveTo(row.statementEdge, row.yPos) +
+        this.insideCornerPaths_.pathTop(this.RTL_) +
+        Blockly.utils.svgPaths.lineOnAxis('v', innerHeight) +
+        this.insideCornerPaths_.pathBottom(this.RTL_);
+    this.steps_.push(steps);
   } else {
-    this.highlightSteps_.push('M',
-        (x + Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE) + ',' +
-        (row.yPos + row.height - Blockly.blockRendering.constants.DISTANCE_45_OUTSIDE));
-    this.highlightSteps_.push(
-        Blockly.blockRendering.constants.INNER_BOTTOM_LEFT_CORNER_HIGHLIGHT_LTR);
+    var steps =
+        Blockly.utils.svgPaths.moveTo(row.statementEdge, row.yPos + row.height) +
+        this.insideCornerPaths_.pathBottom(this.RTL_);
+    this.steps_.push(steps);
   }
 };
 
 Blockly.blockRendering.Highlighter.prototype.drawRightSideRow = function(row) {
   if (row.followsStatement) {
-    this.highlightSteps_.push('H', row.width);
+    this.steps_.push('H', row.width);
   }
-  if (this.info_.RTL) {
-    this.highlightSteps_.push('H', row.width - Blockly.blockRendering.constants.HIGHLIGHT_OFFSET);
-    this.highlightSteps_.push('v', row.height);
+  if (this.RTL_) {
+    this.steps_.push('H', row.width - this.highlightOffset_);
+    this.steps_.push('v', row.height);
   }
 };
 
 Blockly.blockRendering.Highlighter.prototype.drawBottomCorner = function(_row) {
   var height = this.info_.height;
   var elems = this.info_.bottomRow.elements;
 
-  if (this.info_.RTL) {
-    this.highlightSteps_.push('V', height);
+  if (this.RTL_) {
+    this.steps_.push('V', height);
   }
 
   for (var i = elems.length - 1; i >= 0; i--) {
     var elem = elems[i];
     if (elem.type === 'square corner') {
-      if (!this.info_.RTL) {
-        this.highlightSteps_.push('M',
-            Blockly.blockRendering.constants.HIGHLIGHT_OFFSET + ',' +
-            (height - Blockly.blockRendering.constants.HIGHLIGHT_OFFSET));
+      if (!this.RTL_) {
```

Comment:
I think you can skip the loop in RTL because nothing ever gets pushed.

Reply:
Done.  Much simpler now.